File: Labs/Lab4/TreeNode.java
public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
    public static TreeNode invertTree(TreeNode root) { //lab4
        if (root == null) {
            return root;
        }
        if (root.left != null) { //dfs traversal
            invertTree(root.left);
            if (root.right != null) {
                invertTree(root.right);
            }
        }
        TreeNode tmp = root.left;
        root.left = root.right;
        root.right = tmp;
        return root;
    }
}

File: Labs/Lab4/TreeNode.py
class TreeNode(object):
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

    def invertTree(self, root): #lab4
        if root is None:
            return root
        if root.left is not None:
            self.invertTree(root.left)
            if root.right is not None:
                self.invertTree(root.right)
        tmp = root.left
        root.left = root.right
        root.right = tmp
        return root

